lib/SILOptimizer/Transforms/FunctionSignatureOpts.cpp

```diff
@@ -63,6 +63,21 @@ using ArgumentIndexMap = llvm::SmallDenseMap<int, int>;
 //                           Utilties 
 //===----------------------------------------------------------------------===//
 
+/// Find the return value of the function.
+static SILValue findReturnValue(SILFunction *F) {
+  // Find the return BB of F. If we fail, then bail.
+  SILFunction::iterator BB = F->findReturnBB();
+  if (BB == F->end())
+    return SILValue();
+
+  for (auto II = BB->rbegin(), IE = BB->rend(); II != IE; ++II) {
```

Comment:
The ReturnInst must be the last instruction of the block: use BB->getTerminator() instead of the loop